src/main/java/com/cloudbees/jenkins/plugins/bitbucket/client/events/BitbucketCloudPullRequestEvent.java

```diff
@@ -70,15 +70,15 @@ private void reconstructMissingData() {
                     this.pullRequest.getSource().getRepository().setScm(repository.getScm());
                 }
                 if (this.pullRequest.getSource().getRepository().getOwner() == null) {
-                    if (this.pullRequest.getSource().getRepository().getOwnerName().equals(this.pullRequest.getAuthorLogin())) {
+                    if (!this.pullRequest.getSource().getRepository().getOwnerName().equals(repository.getOwnerName())) { // i.e., a fork
                         BitbucketCloudRepositoryOwner owner = new BitbucketCloudRepositoryOwner();
-                        owner.setUsername(this.pullRequest.getAuthorLogin());
-                        owner.setDisplayName(this.pullRequest.getAuthorDisplayName());
+                        owner.setUsername(this.pullRequest.getSource().getRepository().getOwnerName());
+                        owner.setDisplayName(this.pullRequest.getAuthorLogin());
                         if (repository.isPrivate()) {
                             this.pullRequest.getSource().getRepository().setPrivate(repository.isPrivate());
                         }
                         this.pullRequest.getSource().getRepository().setOwner(owner);
-                    } else if (this.pullRequest.getSource().getRepository().getOwnerName().equals(repository.getOwnerName())) {
+                    } else { // origin branch
```

Comment:
Possibly fixes a bug with the handling of webhooks for forked PRs in BB Cloud, though I am not sure how to test that outside a mock context (discovered due to unit test failures): I cannot find any non-test usages of `BitbucketRepository.getOwner` except for code in this `reconstructMissingData` method which _sets_ an `owner` field. Perhaps @stephenc can clarify if he still remembers anything about it (but this was over three years ago).